clang/test/3C/type_params_xfail1.c

```diff
@@ -0,0 +1,28 @@
+// RUN: 3c -addcr -alltypes %s | FileCheck -match-full-lines %s
+
+// XFAIL: *
+
+// This fails because the type variable is used to constrain both calls to `incoming_doubleptr`.
+// To be correct, the usage of a type variable should be independent at each call site.
+
+// adapted from type_params.c
+#include <stddef.h>
+_Itype_for_any(T) void *malloc(size_t size) : itype(_Array_ptr<T>) byte_count(size);
+
+_Itype_for_any(T)
+void incoming_doubleptr(_Array_ptr<T> ptr : itype(_Array_ptr<T>)) {
+  return;
+}
+
+void do_doubleptr(int count) {
+  int **arr = malloc(sizeof(int*) * count);
+  // CHECK: _Array_ptr<_Ptr<int>> arr : count(count) = malloc<_Ptr<int>>(sizeof(int*) * count);
```

Comment:
Just to be clear: This is the rewriting that should happen, but does not? What happens instead?

Reply:
It is rewritten as `_Array_ptr<_Array_ptr<int>> arr`, because of interference from the other function.